release-tools/check-crate-version/src/main.rs

```diff
@@ -0,0 +1,177 @@
+use std::{cmp::max, collections::HashMap, fs, path::PathBuf, process::Command};
+
+use clap::Parser;
+use serde::Deserialize;
+use toml_diff::TomlDiff;
+
+mod download;
+
+use download::download_crate;
+
+const PUBLISH_LIST_PATH: &str = "./publish-list.toml";
+const CRATES_DIR: &str = "../../crates";
+const CRATES_IO_DIR: &str = "./crates_io";
+
+enum CrateStatus {
+    NotPublished,
+    VersionBumped(String),
+    Unchanged,
+    CodeChanged(Option<String>),
+    ManifestChanged(String),
+}
+
+#[derive(Parser, Debug)]
+pub struct Cli {
+    #[clap(short, long)]
+    verbose: bool,
+}
+
+#[tokio::main]
+async fn main() {
+    let cli = Cli::parse();
+    let verbose = cli.verbose;
+
+    let publish_list = read_publish_list();
+    let padding = publish_list
+        .iter()
+        .fold(0, |len, name| max(len, name.len()));
+
+    let status_checks: Vec<_> = publish_list
+        .into_iter()
+        .map(|crate_name| {
+            (
+                crate_name.clone(),
+                tokio::spawn(check_crate_status(crate_name)),
```

Comment:
Can we add a rate limit for these checks so we can abide by crates.io's crawler policy of `1 request per second or less`?

https://crates.io/policies#crawlers

Reply:
Bummer, cause it means there was no point in making this async, but you're right, that is what the crawler policy says. Implemented.